### PR TITLE
Remove support for non-Octane blueprints

### PR DESCRIPTION
**Pull request by cafreeman**
The current blueprints test suite covers cases for both octane and non-octane blueprints, even though non-octane blueprints are no longer supported. As a result, [this PR to upgrade ember-cli to 4.1.0](https://github.com/emberjs/ember.js/pull/19893) is blocked because all of the non-octane blueprint tests fail (since the CLI no longer generates non-Octane files).

This PR unblocks the ember-CLI upgrades by removing all of the non-Octane blueprint tests. In addition, it also removes all of the pre-Octane blueprints and fixtures since they are no longer supported and can't be generated in the current version of Ember CLI.

### CODE REVIEW

**Review by locks — APPROVED** [review]

File: blueprints/component-addon/index.js
Comment (rwjblue) [inline blueprints/component-addon/index.js:7]: I think this file (`blueprints/edition-detector.js`) still needs to be deleted, right?

File: node-tests/blueprints/component-class-test.js
Comment (rwjblue) [inline node-tests/blueprints/component-class-test.js:60]: Can you double check what the RFC says RE: "--component-structure=classic"? I think we probably don't need that flag value anymore...

File: node-tests/blueprints/controller-test.js
Comment (rwjblue) [inline node-tests/blueprints/controller-test.js:122]: Why remove these?

File: node-tests/blueprints/route-test.js
Comment (rwjblue) [inline node-tests/blueprints/route-test.js:1062]: Why are these `ember-page-title` related tests deleted (are they brought back in another file somewhere)?

File: node-tests/blueprints/route-test.js
Comment (rwjblue) [inline node-tests/blueprints/route-test.js:759]: We still need these pods tests

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: node-tests/blueprints/route-test.js
Comment (cafreeman) [reply node-tests/blueprints/route-test.js:1062]: Since `ember-page-title` is included by default now, I deleted these tests and instead added page title-related assertions to *all* of the route tests since all routes generate templates with page title anyway.

File: blueprints/component-addon/index.js
Comment (cafreeman) [reply blueprints/component-addon/index.js:7]: Good catch

File: node-tests/blueprints/controller-test.js
Comment (cafreeman) [reply node-tests/blueprints/controller-test.js:122]: 🤔 looking at this diff, it seems like there were some copy/paste shenanigans going on with the pod tests. gonna take another pass on this file

File: node-tests/blueprints/route-test.js
Comment (bertdeblock) [reply node-tests/blueprints/route-test.js:1062]: Shouldn't we still have tests that verify that the `page-title` helper is not present when `ember-page-title` is not installed? I could be wrong, but it seems all tests now assume the helper is always present?

File: node-tests/blueprints/route-test.js
Comment (cafreeman) [reply node-tests/blueprints/route-test.js:759]: they still exist, they're just higher up in the file https://github.com/cafreeman/ember.js/blob/cfreeman/remove-non-octane-blueprint-tests/node-tests/blueprints/route-test.js#L240-L282

File: node-tests/blueprints/route-test.js
Comment (cafreeman) [reply node-tests/blueprints/route-test.js:1062]: Hmm, I'm not sure. Since `ember-page-title` is installed by default, and these are blueprint tests, the blueprints in a new Ember app *will* always have page-title present unless people go and remove it after the fact. I guess we could still include a test or two for the case where it isn't installed just to make sure that it doesn't break if users remove it.

Comment (bertdeblock) [reply node-tests/blueprints/route-test.js:1062]: I think having at least one test that still verifies the absence of the helper when `ember-page-title` is not installed is a good idea. These blueprints will also be used by people who upgrade their app and might have deleted `ember-page-title` in the past because e.g. they use `ember-cli-head` or a different approach. Just a suggestion though! Someone from the core team should decide.

Comment (rwjblue) [reply node-tests/blueprints/route-test.js:1062]: Ya, agree with @bertdeblock here. If the project doesn't have ember-page-title we need to not add helper invocations in the blueprint.

File: node-tests/blueprints/component-class-test.js
Comment (cafreeman) [reply node-tests/blueprints/component-class-test.js:60]: For anyone referencing this later, after discussing with @rwjblue we decided this would be fine in a separate PR.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**
